api/client_test.go

```diff
@@ -123,3 +123,39 @@ func TestGet(t *testing.T) {
 		assert.DeepEqual(t, req.Header, expectedHeaders)
 	})
 }
+
+func TestDelete(t *testing.T) {
+	ch := make(chan *http.Request, 1)
+	handler := func(rw http.ResponseWriter, r *http.Request) {
+		ch <- r
+		assert.Equal(t, r.Method, http.MethodDelete)
+		assert.Equal(t, r.URL.Path, "/good")
```

Comment:
Since this runs in a different goroutine it's going to panic instead of give a nice error message.

I think these would be better off below next to the `assert.DeepEqual`, so that the failure message is easier to read.